internal/commander.py

```
@@ -0,0 +1,85 @@
+            scale_fix_filter = helpers.ResolutionFixer.CROP
+            vf_list.append(scale_fix_filter)
+
+        # обработка  fps
+        if fps and fps != "don't change":
+            vf_list.append(f"fps={fps}")
+
+        # обработка pixel_format
+        if pixel_format and pixel_format != "don't change":
+            vf_list.append(f"format={pixel_format}")
+
+        vf = f'-vf "{",".join(vf_list)}"' if vf_list else ""
+
+        return vf
+
+    def build_ffmpeg_command(self, file: str, output_dir) -> str:
+        """
+        Генерация ffmpeg команды для одного файла с учётом настроек.
+        Универсальная для любого кодека, использует codec.vcodec/acodec.
+        """
+
+        filename, ext = os.path.splitext(os.path.basename(file))
+        crf = self.settings.get("crf")
+        preset = self.settings.get("preset")
+        container = self.settings.get("container")
+        passes = self.settings.get("passes")
+        audio_bitrate = int(self.settings.get("audio bitrate"))
+
+        vf = self.__build_vf_list()
+
+        output_file = os.path.join(output_dir, f"{filename}_{self.codec.name}_q{crf}")
+
+        vcodec = self.codec.vcodec
+        acodec = self.codec.acodec
+
+        if passes == "One-Pass":
+            cmd = (
+                f'ffmpeg -y -i "{file}" {vf} -c:v {vcodec} '
+                f'-preset {str(preset)} -crf {str(crf)} '
+                f'-c:a {acodec} -b:a {audio_bitrate}k "{output_file}_1pass.{container}"'
+            )
+        elif passes == "Two-Pass":
+            cmd = (
+                f'ffmpeg -y -i "{file}" {vf} -c:v {vcodec} '
+                f'-preset {str(preset)} -crf {str(crf)} -pass 1 -an -f null {os_adapter.NULL_DEVICE} && '
+                f'ffmpeg -y -i "{file}" {vf} -c:v {vcodec} '
+                f'-preset {str(preset)} -crf {str(crf)} -pass 2 '
+                f'-c:a {acodec} -b:a {audio_bitrate}k "{output_file}_2pass.{container}"'
+            )
+        else:
```

ещё один best practice это early fail. такой рейт надо делать в начале метода. да и, вообще, посмотри в своём проекте, может быть где-то ещё не хватает early fail.

"Fail Fast" and "Fail Early" are related software design and agile development principles that advocate for error detection and handling as soon as possible to prevent incorrect states from persisting and causing more significant problems later